File: backend/router/board.go
package router

import (
	"TalkSphere/controller"
	"TalkSphere/middleware"

	"github.com/gin-gonic/gin"
)

func InitBoardRouter(r *gin.RouterGroup) {
	// 公开接口 - 不需要登录
	publicGroup := r.Group("/boards")
	{
		publicGroup.GET("", controller.GetAllBoards) // 获取板块列表不需要登录
	}

	// 需要登录的接口
	authGroup := r.Group("/boards")
	authGroup.Use(middleware.JWTAuthMiddleware(), middleware.AdminRequired())
	{
		authGroup.POST("", controller.CreateBoard)       // 创建板块需要登录
		authGroup.PUT("/:id", controller.UpdateBoard)    // 更新板块需要登录
		authGroup.DELETE("/:id", controller.DeleteBoard) // 删除板块需要登录
	}
}

File: frontend/src/api/board.js
import request from '../utils/request'

// 获取所有板块
export function getAllBoards() {
  return request({
    url: 'api/boards',
    method: 'get'
  })
}

// 创建板块
export function createBoard(data) {
  return request({
    url: 'api/boards',
    method: 'post',
    data
  })
}

// 更新板块
export function updateBoard(id, data) {
  return request({
    url: `/boards/${id}`,
    method: 'put',
    data
  })
}

// 删除板块
export function deleteBoard(id) {
  return request({
    url: `/boards/${id}`,
    method: 'delete'
  })
} 